tablecodec/tablecodec.go

```diff
@@ -469,6 +469,16 @@ func appendTableIndexPrefix(buf []byte, tableID int64) []byte {
 	return buf
 }
 
+// ReplaceRecordKeyTableID replace the tableID in the recordKey buf.
+func ReplaceRecordKeyTableID(buf []byte, tableID int64) []byte {
+	if len(buf) < len(tablePrefix)+8 {
+		return buf
+	}
+
+	_ = codec.PutInt(buf[len(tablePrefix):], tableID)
```

Comment:
How about use `EncodeInt` and `copy`?
Better not adding a new API if the existing ones will do.